src/Cryptography/Sodium/Bindings/Signing.hs

```diff
@@ -0,0 +1,276 @@
+-- into @sm@ and is 'cryptoSignBytes' + @mlen@ bytes long.
+--
+-- If @smlen@ is not a @NULL@ pointer, then the actual length of the signed message is stored in
+-- @smlen@.
+--
+-- @since 0.0.1.0
+foreign import capi "sodium.h crypto_sign"
+  cryptoSign ::
+    -- | @sm@ parameter.
+    Ptr CUChar ->
+    -- | @smlen_p@ parameter.
+    Ptr CULLong ->
+    -- | @m@ parameter.
+    Ptr CUChar ->
+    -- | @mlen@ parameter.
+    CULLong ->
+    -- | @sk@ parameter.
+    Ptr CUChar ->
+    -- | Return code is 0 on success, -1 on error.
+    IO CInt
+
+-- | The 'cryptoSignOpen' function checks that the signed message @sm@,
+-- whose length is @smlen@ bytes, has a valid signature for the public key @pk@.
+--
+-- On success, it puts the message without the signature into @m@, stores its length in @mlen@
+-- if @mlen@ is not a @NULL@ pointer.
+--
+-- @since 0.0.1.0
+foreign import capi "sodium.h crypto_sign_open"
+  cryptoSignOpen ::
+    -- | @m@ parameter.
+    Ptr CUChar ->
+    -- | @mlen_p@ parameter
+    Ptr CULLong ->
+    -- | @sm@ parameter.
+    Ptr CUChar ->
+    -- | @smlen@ parameter.
+    CULLong ->
+    -- | @pk@ parameter.
+    Ptr CUChar ->
+    -- | On success, the function returns 0
+    -- If the signature isn't valid, then the function returns -1.
+    IO CInt
+
+-------------------
+-- Detached Mode --
+-------------------
+
+-- | The 'cryptoSignDetached' function signs the message @m@, whose length is @mlen@ bytes,
+-- using the secret key @sk@ and puts the signature into @sig@, which can be up to
+-- 'cryptoSignBytes' bytes long.
+-- The actual length of the signature is put into @siglen@ if @siglen@ is not @NULL@.
+-- It is safe to ignore @siglen@ and always consider a signature as 'cryptoSignBytes' bytes long;
+-- shorter signatures will be transparently padded with zeros if necessary.
+--
+-- @since 0.0.1.0
+foreign import capi "sodium.h crypto_sign_detached"
+  cryptoSignDetached ::
+    -- | @sig@ parameter.
+    Ptr CUChar ->
+    -- | @siglen_p@.
+    Ptr CULLong ->
+    -- | @m@ parameter.
+    Ptr CUChar ->
+    -- | @mlen@ parameter.
+    CULLong ->
+    -- | @sk@ parameter.
+    Ptr CUChar ->
+    -- | Return code is 0 on success, -1 on error.
+    IO CInt
+
+-------------------------
+-- Multi-part messages --
+-------------------------
+
+-- $mpm
+-- If the message you're trying to sign doesn't fit in memory, then it can be provided as a sequence of arbitrarily-sized
+-- chunks. This uses the @Ed25519ph@ signature system, which pre-hashes the message. In other words,
+-- what gets signed is not the message itself but its image through a hash function. If the message
+-- can fit in memory and be supplied as a single chunk, then the single-part API should be
+-- preferred. Note: @Ed25519ph(m)@ is intentionally not equivalent to Ed25519(SHA512(m)). If, for
+-- some reason, you need to pre-hash the message yourself, then use the multi-part
+-- @crypto_generichash_*()@ APIs and sign the 512-bit output.
```

Comment:
Use the Haskell equivalent - referring to C functions in this way in Haddocks makes very little sense.